File: app/src/main/java/app/wallpaper/app/ApplicationLoader.java
package app.wallpaper.app;

import android.app.Application;
import android.content.Context;
import android.os.Handler;
import android.os.Looper;


public final class ApplicationLoader extends Application {

    public static Context applicationContext;
    public static Handler handler;

    @Override
    public void onCreate() {
        super.onCreate();

        applicationContext = this;
        handler = new Handler(Looper.getMainLooper());
    }
}

File: app/src/main/java/app/wallpaper/app/ApplicationLoader.kt
package app.wallpaper.app

import android.content.Context
import app.wallpaper.di.DaggerAppComponent
import dagger.android.AndroidInjector
import dagger.android.DaggerApplication

class ApplicationLoader : DaggerApplication() {

    private lateinit var injector: AndroidInjector<out DaggerApplication>

    override fun onCreate() {
        super.onCreate()
        APPLICATION = this
    }

    override fun attachBaseContext(base: Context?) {
        super.attachBaseContext(base)

        injector = DaggerAppComponent.builder()
            .application(this)
            .build()
    }

    override fun applicationInjector(): AndroidInjector<out DaggerApplication> {
        return injector
    }

    companion object {

        private lateinit var APPLICATION: ApplicationLoader

        fun get(): ApplicationLoader = APPLICATION
    }
}